python/pyarrow/tests/test_flight.py

```diff
@@ -862,3 +862,38 @@ def test_roundtrip_errors():
             list(client.do_action(flight.Action("unauthorized", b"")))
         with pytest.raises(flight.FlightInternalError, match=".*foo.*"):
             list(client.list_flights())
+
+
+def test_do_put_independent_read_write():
+    """Ensure that separate threads can read/write on a DoPut."""
+    # ARROW-6063: previously this would cause gRPC to abort when the
```

Comment:
I'll ask my question again: is it possible to add our own checks and return an error in that case, instead of letting gRPC abort?

Reply:
IIUC, the error condition is "one of reader or writer has been closed and we're trying to issue operations on the other one".

Reply:
If it's just a flag, an atomic bool would probably suffice.